operator/pkg/helmreconciler/prune.go

```diff
@@ -63,6 +63,7 @@ var (
 		{Group: "admissionregistration.k8s.io", Version: "v1", Kind: name.ValidatingWebhookConfigurationStr},
 		{Group: "rbac.authorization.k8s.io", Version: "v1", Kind: name.ClusterRoleStr},
 		{Group: "rbac.authorization.k8s.io", Version: "v1", Kind: name.ClusterRoleBindingStr},
+		{Group: "gateway.networking.k8s.io", Version: "v1", Kind: name.GatewayClassStr},
```

Comment:
How does this work? I think they are created by pilot, and istioctl is unable to delete these resources?

Reply:
``` shell
# kubectl get gatewayclass
NAME           CONTROLLER                                      ACCEPTED   AGE
istio          istio.io/gateway-controller                     True       32s
istio-remote   istio.io/unmanaged-gateway                      True       32s

```

I'm not sure what created these two resources, but my understanding is that --purge should delete them. Am I right?


I'm testing whether my PR can correctly delete these two resources.